services/src/main/java/org/keycloak/protocol/oidc/endpoints/TokenEndpoint.java

```diff
@@ -678,6 +678,7 @@ public Response resourceOwnerPasswordCredentialsGrant() {
         AuthenticationProcessor processor = new AuthenticationProcessor();
         processor.setAuthenticationSession(authSession)
                 .setFlowId(flowId)
+                .setFlowPath("token")
```

Comment:
Looking at https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java#L544, we should be doing something very similar as you are doing in the `FreemarkerLoginFormsProvider`. If the request is a backend request we don't really need to handle forms.

@mposolda wdyt?